server/api/api_accessor.py

```diff
@@ -8,53 +7,56 @@
 from server.config import (
     API_BASE_URL, API_CLIENT_ID, API_CLIENT_SECRET, API_TOKEN_URI
 )
+from server.decorators import with_logger
 
 from .oauth_session import OAuth2Session
 
-"""
-Uncomment the following line if your API uses HTTP instead of HTTPS
-"""
-# os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'
-
 
+@with_logger
 class SessionManager:
     """
     Garantor for API access
     """
     def __init__(self):
         self.session = None  # Instance of session
-        self.logger = logging.getLogger()
 
     async def get_session(self) -> Optional[OAuth2Session]:
-        if self.session:
-            return self.session
-
-        try:
+        if not self.session:
```

Comment:
Is there more than just the `is not None` check here?

Reply:
No